test/conftest.py

```diff
@@ -3,227 +3,23 @@
-    """
-    This fixture returns current build mode.
-    This is for running tests through the test.py script, where only one mode is passed to the test
-    """
-    # to avoid issues when there's no provided mode parameter, do it in two steps: get the parameter and if it's not
-    # None, get the first value from the list
-    mode = request.config.getoption("modes")
-    if mode:
-        return mode[0]
-    return "unknown"
-
-@pytest.fixture(scope="function")
-def get_params(request: pytest.FixtureRequest) -> Generator[None]:
-    # this dummy fixture only needed to modify the test name with run id and mode. We don't want to parametrize with
-    # some parameters, so we are returning existing params that function is accepting. This method only needed for
-    # pytest_generate_tests method
-    return request.param
-
-
-def pytest_generate_tests(metafunc: pytest.Metafunc) -> None:
-    if TEST_RUNNER == "runpy":
-        return
-    repeat_count = metafunc.config.getoption("--repeat")
-    modes = get_modes_to_run(metafunc.config)
-
-    all_combinations = [*itertools.product(modes, range(repeat_count))]
-
-    metafunc.fixturenames.append('get_params')
-    metafunc.parametrize(
-        'get_params',
-        range(len(all_combinations)),
-        ids=[f"%{mode}.{run_id + 1}%" for mode, run_id in all_combinations],
-        indirect=True
-    )
-
-@pytest.fixture(autouse=True)
-def print_scylla_log_filename(request: pytest.FixtureRequest) -> Generator[None]:
-    """Print out a path to a ScyllaDB log.
 
-    This is a fixture for Python test suites, because they are using a single node clusters created inside test.py,
-    but it is handy to have this information printed to a pytest log.
-    """
 
-    yield
+pytest_plugins = []
 
-    if scylla_log_filename := request.config.getoption("--scylla-log-filename"):
-        logger.info("ScyllaDB log file: %s", scylla_log_filename)
 
-
-def testpy_test_fixture_scope(fixture_name: str, config: pytest.Config) -> str:
-    """Dynamic scope for fixtures which rely on a current test.py suite/test.
-
-    test.py runs tests file-by-file as separate pytest sessions, so, `session` scope is effectively close to be the
-    same as `module` (can be a difference in the order.)  In case of running tests with bare pytest command, we
-    need to use `module` scope to maintain same behavior as test.py, since we run all tests in one pytest session.
-    """
-    if config.getoption("--test-py-init"):
-        return "module"
-    return "session"
-
-testpy_test_fixture_scope.__test__ = False
-
-
-@pytest.fixture(scope=testpy_test_fixture_scope)
-async def testpy_test(request: pytest.FixtureRequest, build_mode: str) -> Test | None:
-    """Create an instance of Test class for the current test.py test."""
-
-    if request.scope == "module":
-        return await get_testpy_test(path=request.path, options=request.config.option, mode=build_mode)
-    return None
+if TEST_RUNNER == "runpy":
```

Comment:
Wow, until now I didn't notice that in commit d0e4045103d you stuck a os.environ.setdefault('SCYLLA_TEST_RUNNER', 'runpy')

Wow, this was completely wrong. I'll open a separate issue about that.